.circleci/config.yml

```diff
@@ -353,6 +353,11 @@ jobs:
 
 workflows:
   main:
+    when:
+      or:
+        - equal: [build_four_hours, << pipeline.trigger_source >>]
+        - not:
+            equal: [scheduled_pipeline, << pipeline.trigger_source >>]
```

Comment:
The `when` condition uses an `or` statement to run the workflow in two scenarios:
1. When the trigger source is 'build_four_hours'
1. When the trigger source is not a scheduled pipeline (which covers all branch pushes and pull requests)